src/App.js

```diff
@@ -19,42 +19,65 @@ import UserProfilePage from './pages/UserProfilePage';
 import { useAuth0 } from '@auth0/auth0-react';
 import { Socket } from './components/ChatService/Socket';
 import ChatPage from './components/ChatService/ChatPage';
+import axios from 'axios';
+
+const axiosInstance = axios.create({
+  baseURL: "http://localhost:5000",
+  headers: {
+    "Content-Type": "application/json"
+  }
+});
 
 export default function App () {
-  const {isAuthenticated, isLoading, user} = useAuth0();
+  const {isAuthenticated, isLoading, user, logout} = useAuth0();
   const [userRole, setUserRole] = useState();
   const [dbUser, setDbUser] = useState();
-  // const [isAuthenticated, setIsAuthenticated] = useState();
 
   useEffect(() => {
     if (!isLoading && isAuthenticated) {
-      const requestOptions = {
-        method: 'GET',
-      };
-      fetch(`http://localhost:5000/get_user_role?id=${user.sub}&email=${user.email}`, requestOptions)
-          .then(response => response.json())
-          .then(data => setUserRole(data.role[0]));
-    }
-  }, [user, isAuthenticated, isLoading])
+      // start session on BE
+      axiosInstance.get('/login', {
+        params: {
+          email: user.email
+        }})
+        .catch(function(error) {
+          logout({
+            returnTo: window.location.origin,
+          });
+          console.log(error);
+        });
+
+      axiosInstance.get('/get_user_role', {
+        params: {
+          id: user.sub,
+        }
+        }).then(response => {
+          setUserRole(response?.data["role"][0]);
+        })
+        .catch(function(error) {
+          console.log(error);
+        });
+
+        axiosInstance.get('/users', {
+          params: {
+            email: user.email,
+          }
+        }).then(response => {
```

Comment:
I meant to fix this last night. Just move this `then` to the call to login, it returns the dbUser

Reply:
ignore me